### PR TITLE
Use format: uri

### LINKED ISSUE #3235: OpenAPI spec doesn't label URI properties
Some `type: string` properties are specified to contain URLs ([avatar_url](https://github.com/matrix-org/matrix-doc/blob/master/data/api/client-server/profile.yaml#L130), [Pusher url](https://github.com/matrix-org/matrix-doc/blob/master/data/api/client-server/pusher.yaml#L221), [well-known base_url](https://github.com/matrix-org/matrix-doc/tree/master/data/api/client-server/definitions/wellknown), [og:image](https://github.com/matrix-org/matrix-doc/blob/master/data/api/client-server/content-repo.yaml#L390)), but this isn't machine readable.
JSON Schema allows to [specify URL strings](https://json-schema.org/understanding-json-schema/reference/string.html#resource-identifiers) with the formats uri, uri-reference, iri, and iri-reference. [OpenAPI v2](https://swagger.io/specification/v2/#data-types) doesn't define these, so the JSON schema definitions apply. The questions which format to use are:
- Do these URLs have to be absolute? I suppose so, because the base url would be unclear, so let's not use the -reference formats.
- Are raw non-ASCII characters allowed in these URLs? I would favor it, so let's use the iri format.

Do you agree with the iri format? Would you accept a PR that adds an URL format to these properties?

### PR DESCRIPTION
**Pull request by lukaslihotzki**
Fix #3235

### CODE REVIEW

**Comment by lukaslihotzki** [comment]
@turt2live, @KitsuneRal: Can you review this?

File: changelogs/client_server/newsfragments/3236.clarification [hunk 1]
```diff
@@ -0,0 +1 @@
+Use uri format in avatar_url, pusher url, wellknown base_url, getUrlPreview, and submit_url.
```
Comment (turt2live) [inline changelogs/client_server/newsfragments/3236.clarification:1]: We don't actually have a changelog for the swagger, so this PR is fine to exclude the entry.

**Review by turt2live — APPROVED** [review]
otherwise looks good - thanks!

**Review by KitsuneRal — CHANGES_REQUESTED** [review]
While updating libQuotient to the code generated from this PR I found this place needs `format: uri` too: https://github.com/matrix-org/matrix-doc/pull/3236/files#diff-290a7b6086f893635e8372140dfde8ca9ab1040e07cfa2bc0c50c14e7feb8879R67

**Comment by lukaslihotzki** [comment]
@KitsuneRal: Which place needs `format: uri` too? The link points to data/api/client-server/definitions/public_rooms_response.yaml, line 67, which is the `guest_can_join` boolean property.

**Comment by KitsuneRal** [comment]
> @KitsuneRal: Which place needs `format: uri` too? The link points to data/api/client-server/definitions/public_rooms_response.yaml, line 67, which is the `guest_can_join` boolean property.

Huh, that's not what it was supposed to point to. The piece in question is in `data/api/client-server/content-repo.yaml` around line 67 - the `content_uri` field returned by `/upload`.

**Review by KitsuneRal — APPROVED** [review]

**Pull request merged**